search_data.json

```diff
@@ -0,0 +1,50 @@
+---
+layout: null
+---
+{
+  {% for item_hash in data.resource %}
+    {% assign object = item_hash[1] %}
+    {% assign id = item_hash[0] %}
+    {% assign collection_id = object.uri | split:'/' | last %}
+    {% assign subjects = "" | split: ',' %}
+    {% assign agents = "" | split: ',' %}
+    {% assign dates = "" | split: ',' %}
+
+    {% for date in object.dates %}
+      {% assign dates = dates | push: date.expression %}
+    {% endfor %}
+
+    {% for subject in object.subjects %}
+      {% capture subject_id %}{{ subject.ref | split: '/' | last }}{% endcapture %}
+      {% assign obj = site.data.subject[subject_id] %}
+      {% assign subjects = subjects | push: obj.title %}
+    {% endfor %}
+
+    {% for agent in object.linked_agents %}
+      {% capture agent_id %}{{ agent.ref | split: '/' | last }}{% endcapture %}
+      {% if agent.ref contains "corporate_entities" %}
+        {% assign agent_data = site.data.agent_corporate_entity[agent_id] %}
+        {% assign agents = agents | push: agent_data.title %}
+      {% elsif agent.ref contains "people" %}
+        {% assign agent_data = site.data.agent_person[agent_id] %}
+        {% assign agents = agents | push: agent_data.title %}
+      {% endif %}
+    {% endfor %}
+
+    {% for instance in object.instances %}
+      {%- capture top_container_id -%}
+        {{ instance.sub_container.top_container.ref | split: '/' | last }}
+      {%- endcapture -%}
+      {% assign obj = site.data.top_container[top_container_id] %}
+    {% endfor %}
+
+    "resource/{{id}}": {
+      "href": "resource/{{id}}",
+      "title": "{{object.title | escape | strip_newlines | remove: '\'}}",
+      "url": "{{site.url}}/resource/{{id}}",
+      "dates": "{{ dates }}",
+      "agents": "{{ agents }}",
+      "subjects": "{{ subjects }}",
```

Comment:
You'll need to do something so you have a space in between items in the array.  See https://shopify.github.io/liquid/filters/join/